flink-cdc-connect/flink-cdc-pipeline-connectors/flink-connector-test-cdc/src/main/java/com/ververica/cdc/connectors/test/ValuesDatabase.java

```diff
@@ -0,0 +1,391 @@
+        // [primaryKeys, [column_name, column_value]]
+        private final Map<String, Map<String, String>> records;
+
+        private final LinkedList<Column> columns;
+
+        private final List<String> primaryKeys;
+
+        // indexes of primaryKeys in columns
+        private final List<Integer> primaryKeyIndexes;
+
+        public ValuesTable(TableId tableId, Schema schema) {
+            this.tableId = tableId;
+            this.lock = new Object();
+            this.columns = new LinkedList<>(schema.getColumns());
+            this.records = new HashMap<>();
+            this.primaryKeys = new LinkedList<>(schema.primaryKeys());
+            this.primaryKeyIndexes = new ArrayList<>();
+            updatePrimaryKeyIndexes();
+        }
+
+        public List<String> getResult() {
+            List<String> results = new ArrayList<>();
+            synchronized (lock) {
+                records.forEach(
+                        (key, record) -> {
+                            StringBuilder stringBuilder = new StringBuilder(tableId.toString());
+                            stringBuilder.append(":");
+                            for (Column column : columns) {
+                                stringBuilder
+                                        .append(column.getName())
+                                        .append("=")
+                                        .append(record.getOrDefault(column.getName(), ""))
+                                        .append(";");
+                            }
+                            stringBuilder.deleteCharAt(stringBuilder.length() - 1);
+                            results.add(stringBuilder.toString());
+                        });
+            }
+            return results;
+        }
+
+        public void applyDataChangeEvent(DataChangeEvent event) {
+            synchronized (lock) {
+                switch (event.op()) {
+                    case INSERT:
+                        {
+                            insert(event.after());
+                            break;
+                        }
+                    case DELETE:
+                        {
+                            delete(event.before());
+                            break;
+                        }
+                    case REPLACE:
+                    case UPDATE:
+                        {
+                            update(event.before(), event.after());
+                            break;
+                        }
+                }
+            }
+        }
+
+        private void delete(RecordData recordData) {
+            records.remove(buildPrimaryKeyStr(recordData));
+        }
+
+        private void insert(RecordData recordData) {
+            String primaryKey = buildPrimaryKeyStr(recordData);
+            Map<String, String> record = new HashMap<>();
+            for (int i = 0; i < recordData.getArity(); i++) {
+                record.put(columns.get(i).getName(), recordData.getString(i).toString());
```

Comment:
Not every field can convert String, could we consider use generic data structure according the Schema instead of String type, this may hide many type conversion and schema conversion issue